Too restrictive typescript type on bot.url function

## Context

My goal is to have a function that gets called when the user sends a message with a url with a *specific* domain in it.

So if the user sends any of the following then I want my function to run:
```
example.com
https://example.com
https://example.com/my-sub-page
```

I'm new to the library but it looks like `bot.url` does what I want it to (although it's not really documented). But because the first parameter is labed `MaybeArray<string>`, typescript throws an error when I supply regex instead of a string.

It looks like there's a duplicate issue here: #1111

* Telegraf.js Version: 4.0.3
* Node.js Version: v14.4.0
* Operating System: OSX

## Minimal Example Code Reproducing the Issue

```ts
import { Telegraf } from 'telegraf'

const token = process.env.BOT_TOKEN;

if (token === undefined) {
  throw new Error('BOT_TOKEN must be provided!')
}

const bot = new Telegraf(token);

// @ts-ignore: MaybeArray<string> can also be a regex under the hood
bot.url(/^.example.com.*$/, (ctx) => {
    ctx.reply(`With match: ${ctx.match}`);
});

bot.launch()
```

If you remove the comment with `@ts-ignore`, the compiler will error with:
```bash
$ npx tsc                                                                                                                                                       127 ↵
src/index.ts:11:9 - error TS2345: Argument of type 'RegExp' is not assignable to parameter of type 'string | string[]'.
  Type 'RegExp' is missing the following properties from type 'string[]': length, pop, push, concat, and 28 more.

11 bot.url(/^.example.com.*$/, (ctx) => {
           ~~~~~~~~~~~~~~~~~~


Found 1 error.
```

--- a/src/composer.ts
+++ b/src/composer.ts
@@ -183,19 +183,19 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
     return this.use(Composer.entity<C, T>(predicate, ...fns))
   }
 
-  email(email: MaybeArray<string>, ...fns: MatchedMiddleware<C>) {
+  email(email: Triggers<C>, ...fns: MatchedMiddleware<C>) {
     return this.use(Composer.email<C>(email, ...fns))
   }
 
-  url(url: MaybeArray<string>, ...fns: MatchedMiddleware<C>) {
+  url(url: Triggers<C>, ...fns: MatchedMiddleware<C>) {
     return this.use(Composer.url<C>(url, ...fns))
   }
 
-  textLink(link: MaybeArray<string>, ...fns: MatchedMiddleware<C>) {
+  textLink(link: Triggers<C>, ...fns: MatchedMiddleware<C>) {
     return this.use(Composer.textLink<C>(link, ...fns))
   }
 
-  textMention(mention: MaybeArray<string>, ...fns: MatchedMiddleware<C>) {
+  textMention(mention: Triggers<C>, ...fns: MatchedMiddleware<C>) {
     return this.use(Composer.textMention<C>(mention, ...fns))
   }
 
@@ -203,7 +203,7 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
     return this.use(Composer.mention<C>(mention, ...fns))
   }
 
-  phone(number: MaybeArray<string>, ...fns: MatchedMiddleware<C>) {
+  phone(number: Triggers<C>, ...fns: MatchedMiddleware<C>) {
     return this.use(Composer.phone<C>(number, ...fns))
   }
 
@@ -493,35 +493,35 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
   }
 
   static email<C extends Context>(
-    email: MaybeArray<string>,
+    email: Triggers<C>,
     ...fns: MatchedMiddleware<C>
   ): MiddlewareFn<C> {
     return Composer.entityText<C>('email', email, ...fns)
   }
 
   static phone<C extends Context>(
-    number: MaybeArray<string>,
+    number: Triggers<C>,
     ...fns: MatchedMiddleware<C>
   ): MiddlewareFn<C> {
     return Composer.entityText<C>('phone_number', number, ...fns)
   }
 
   static url<C extends Context>(
-    url: MaybeArray<string>,
+    url: Triggers<C>,
     ...fns: MatchedMiddleware<C>
   ): MiddlewareFn<C> {
     return Composer.entityText<C>('url', url, ...fns)
   }
 
   static textLink<C extends Context>(
-    link: MaybeArray<string>,
+    link: Triggers<C>,
     ...fns: MatchedMiddleware<C>
   ): MiddlewareFn<C> {
     return Composer.entityText<C>('text_link', link, ...fns)
   }
 
   static textMention<C extends Context>(
-    mention: MaybeArray<string>,
+    mention: Triggers<C>,
     ...fns: MatchedMiddleware<C>
   ): MiddlewareFn<C> {
     return Composer.entityText<C>('text_mention', mention, ...fns)

